Nightly Dotty workflow of 2024-11-04 failed

See https://github.com/scala/scala3/actions/runs/11657389191

```
Error:  Test dotty.tools.dotc.profile.ChromeTraceTest.traceDurationEvents failed: java.lang.AssertionError: null, took 0.041 sec
Error:      at dotty.tools.dotc.profile.ChromeTraceTest$$anon$3.applyOrElse(ChromeTraceTest.scala:87)
Error:      at dotty.tools.dotc.profile.ChromeTraceTest$$anon$3.applyOrElse(ChromeTraceTest.scala:63)
Error:      at dotty.tools.dotc.profile.ChromeTraceTest.testTraceOutputs(ChromeTraceTest.scala:21)
Error:      at dotty.tools.dotc.profile.ChromeTraceTest.traceDurationEvents(ChromeTraceTest.scala:91)
Error:      ...
```

...but it seems it's flaky...

```
Error:  All tests passed after retry, indicating that some tests are flaky.
Error:  This build is configured to reject flaky tests.
Error:  Failed: Total 672, Failed 0, Errors 0, Passed 672, Ignored 8
Error:  (scala3-compiler-bootstrapped / Test / testOnly) sbt.TestsFailedException: Tests unsuccessful
Error:  Total time: 1937 s (32:17), completed Nov 4, 2024 4:46:52 AM
```

cc @hamzaremmal @WojciechMazur any clue?

--- a/compiler/test/dotty/tools/dotc/profile/ChromeTraceTest.scala
+++ b/compiler/test/dotty/tools/dotc/profile/ChromeTraceTest.scala
@@ -54,11 +54,11 @@ class ChromeTraceTest:
     val testStart = System.nanoTime()
     testTraceOutputs{ tracer =>
     tracer.traceDurationEventStart(cat = "test1", name = "event1")
-    LockSupport.parkNanos(2.millis.toNanos)
+    sleep(2.millis)
     tracer.traceDurationEventStart(cat = "test2", name = "event2", colour = "RED", pidSuffix = "pid-suffix")
-    LockSupport.parkNanos(4.millis.toNanos)
+    sleep(4.millis)
     tracer.traceDurationEventEnd(cat = "test2", name = "event2")
-    LockSupport.parkNanos(8.millis.toNanos)
+    sleep(8.millis)
     tracer.traceDurationEventEnd(cat = "test1", name = "event1", colour = "RED", pidSuffix = "pid-suffix")
   }{
     case """{"traceEvents":[""" ::
@@ -89,5 +89,18 @@ class ChromeTraceTest:
           assertTrue(ts4 >= ts3 + 8.millis.toMicros)
         case _ => fail("unreachable")
       }
+    }
   }
-}
+
+  private def sleep(duration: FiniteDuration): Unit = {
+    // A bit of additional precautions to ensure we don't continue execution to early
+    // Both LockSuppport and Thread.sleep can return earlier then expected (depending on OS)
+    var remainingNanos = duration.toNanos
+    val deadline = System.nanoTime() + remainingNanos
+    while
+      remainingNanos = deadline - System.nanoTime()
+      remainingNanos > 0
+    do
+      val millis = NANOSECONDS.toMillis(remainingNanos)
+      Thread.sleep(millis, (remainingNanos % 1.millis.toNanos).toInt)
+  }

